tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+#include "OMPLTestHelpers.hpp"
+#include "../../constraint/MockConstraints.hpp"
+#include <aikido/planner/ompl/Planner.hpp>
+#include <aikido/planner/ompl/CRRT.hpp>
+#include <aikido/planner/ompl/CRRTConnect.hpp>
+#include <aikido/trajectory/Interpolated.hpp>
+#include <aikido/constraint/uniform/RnBoxConstraint.hpp>
+#include <aikido/constraint/CartesianProductSampleable.hpp>
+#include <aikido/constraint/CartesianProductTestable.hpp>
+#include <aikido/constraint/JointStateSpaceHelpers.hpp>
+#include <aikido/util/StepSequence.hpp>
+#include <aikido/planner/ompl/MotionValidator.hpp>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+
+using StateSpace = aikido::statespace::dart::MetaSkeletonStateSpace;
+using aikido::planner::ompl::getSpaceInformation;
+using aikido::planner::ompl::CRRT;
+using aikido::planner::ompl::CRRTConnect;
+
+namespace {
+
+  template <typename T>
+  double computeTrajLength(T& traj,
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+      aikido::distance::DistanceMetricPtr& dmetric)
+  {
+    auto stateCurrent = stateSpace->createState();
+    auto stateNext = stateSpace->createState();
+
+    double trajDistance = 0.0;
+    
+    for(size_t i = 0; i < traj->getDuration(); ++i)
+    {
+      traj->evaluate(i, stateCurrent);   
+      traj->evaluate(i+1, stateNext);
```

Comment:
Not if the code is working correctly. But: (1) a function shouldn't SEGFAULT [which this could] if you provide it a valid input, (2) if an empty trajectory is not a valid input it should be documented and tested as such, and (3) a bug in the code we're testing could easily return an empty trajectory.